src/scripts/eastern_kingdoms/stranglethorn_vale/zulgurub/boss_mandokir.cpp

```diff
@@ -261,6 +252,29 @@ struct boss_mandokirAI : public ScriptedAI
         }
     }
 
+    void CheckVilebranchState(bool reset = false)
+    {
+        // If Vilebranch dies and group wipes, boss should start at the bottom of the stairs
+        // Video: https://www.youtube.com/watch?v=LYq8w4zy3jQ
+        Creature* vileBranch = m_creature->FindNearestCreature(11391, 100.0f, true);
+        bool isVilebranchDead = !vileBranch || !vileBranch->isAlive();
+        if (reset || m_VilebranchDead != isVilebranchDead) {
+            if (isVilebranchDead)
+            {
+                m_creature->RemoveFlag(UNIT_FIELD_FLAGS, START_FLAGS);
+                m_creature->GetMotionMaster()->MoveTargetedHome();
+                m_creature->GetMotionMaster()->MovePoint(0, -12195.0f, -1948.0f, 130.0f, MOVE_NONE, 0.0f, 3.14f);
```

Comment:
Why start Home movement and then immediately overwrite it with Point movement? Seems more logical to first set home position, and then call only MoveTargetedHome().

Reply:
fair enough. i was just copying what was already in the script from UpdateAI. not yet too familiar with the api.